test/models/job_execution_test.rb

```diff
@@ -424,16 +422,24 @@ def with_hidden_errors
 
   describe "#pid" do
     it "returns current pid" do
+      job.command = 'sleep 0.5'
       execution = JobExecution.new('master', job)
-      execution.executor.stubs(pid: 2)
-      execution.pid.must_equal 2
+      JobExecution.start_job(execution)
+      sleep 0.4
+      execution.pid.wont_equal nil
+      execution.wait!
     end
   end
 
   describe "#make_tempdir" do
+    # the actual issue we saw was Errno::ENOTEMPTY ... but that is harder to reproduce
     it "does not fail when directory cannot be removed" do
-      Dir.expects(:rmdir).at_least_once.raises(Errno::ENOTEMPTY)
-      execution.send(:make_tempdir) { 111 }.must_equal 111
+      Airbrake.expects(:notify).with { |e| e.must_include "Notify: make_tempdir error No such" }
```

Comment:
... I guess `.with(matching(xxx))` or something like that could also work ... but this gives a nice error message when it fails